src/hotspot/cpu/x86/c1_LIRAssembler_x86.cpp

```diff
@@ -898,8 +824,7 @@ void LIR_Assembler::reg2mem(LIR_Opr src, LIR_Opr dest, BasicType type, LIR_Patch
       // a dtrace runtime call. This can't work on 64 bit with
       // compressed klass ptrs: T_METADATA can be a compressed klass
       // ptr or a 64 bit method pointer.
-      LP64_ONLY(ShouldNotReachHere());
-      __ movptr(as_Address(to_addr), src->as_register());
+      ShouldNotReachHere();
```

Comment:
Alternatively, you could drop the whole `T_METADATA` case and defer the handling to default case.

Reply:
I initially thought leaving the comment there as meaningful, but now I think that comment only relates to 32-bit x86, so now is redundant. So I dropped the `T_METADATA` case completely.